### PR TITLE
Adding h2 in-memory database

### PR DESCRIPTION
**Pull request by mcruzdev**
Hello @salaboy,

After some days I created the PR from issue #2 

I added the h2 in-memory database and I changed the application.properties file

After I think to implement a repository/DAO layer to remove the in-memory storage like a monolith.

### CODE REVIEW

**Comment by mcruzdev** [comment]
@salaboy I will extract **ConfererenceSiteUtilController** and **ConferenceSiteController** to separate the resposability from class.

**Comment by salaboy** [comment]
@mcruzdev sorry for the miss communication, but yes.. please add the repository/dao layer here.. so someone is using H2, you can just keep commiting to your branch to add commits to this PR

**Review by salaboy — COMMENTED** [review]
@mcruzdev awesome stuff.. can I ask you for one more thing to be added here? can we add some tests for the common endpoints? those are: newProposal, deleteProposal and decision. 
It will be great if we had some tests to check the outputs for those methods.

File: src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Test
+    public void newProposal_ShouldBeCreateAProposal() {
+
+        // action
+        var responseSpec = createProposalRequest();
+
+        // assert
+        responseSpec.expectBody(Proposal.class);
```
Comment (salaboy) [inline src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java]: @mcruzdev can you assert that the proposal has an ID as well? As the backend should create the ID

File: src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    @MockBean(answer = RETURNS_DEEP_STUBS)
+    private ZeebeClientBuilder zeebeClientBuilder;
+
+    private WebTestClient.ResponseSpec createProposalRequest() {
+
+        var requestProposal =
+                new Proposal("Title", "Description", "Author", "email@email.com");
+
+        Mockito.when(proposalRepository.save(requestProposal)).thenReturn(requestProposal);
```
Comment (salaboy) [inline src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java]: @mcruzdev do you really need to mock the DB with h2?

File: src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        createProposalRequest();
+
+        // action, assert
+        webTestClient.delete()
+                .uri("/")
+                .exchange()
+                .expectStatus().isOk();
+
+        getAllProposals().expectBodyList(Proposal.class).hasSize(0);
+    }
```
Comment (salaboy) [inline src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java:143]: @mcruzdev we should have a test here for the decision endpoint.. that is the most tricky one.. can you please add that one?

**Review by salaboy — COMMENTED** [review]
@mcruzdev I've added two small comments and one big about testing the decision endpoint. This is looking good.. one thing that you will need to do for major open-source projects, is to squash all the commits and describe in detail what the commit is adding. Feel free to try this out here..

**Comment by mcruzdev** [comment]
In my last commit, I implemented tests for decision route.
Removed direct mutability from Proposal model, and implemented two methods to be the domain intention clean
Now the test code check the proposal's id

**Review by salaboy — APPROVED** [review]
@mcruzdev great stuff... I will merge now.. and I will be updating the build system

**Pull request merged**